EDMCLogging.py

```diff
@@ -284,7 +284,25 @@ class if relevant.
         caller_qualname = caller_class_names = ''
         if frame:
             # <https://stackoverflow.com/questions/2203424/python-how-to-retrieve-class-information-from-a-frame-object#2220759>
-            frame_info = inspect.getframeinfo(frame)
+            try:
+                frame_info = inspect.getframeinfo(frame)
+                raise(IndexError)  # TODO: Remove, only for testing
+
+            except Exception:
+                # Separate from the print below to guarantee we see at least this much.
+                print('EDMCLogging:EDMCContextFilter:caller_attributes(): Failed in `inspect.getframinfo(frame)`')
+
+                # We want to *attempt* to show something about the nature of 'frame',
+                # but at this point we can't trust it will work.
+                try:
+                    print(f'frame: {frame}')
+
+                except Exception:
+                    pass
+
+                # We've given up, so just return all '??' to signal we couldn't get the info
+                return '??', '??', '??'
```

Comment:
Somewhere in here we should dump the (textual) traceback as well, if we can

Reply:
As discussed on Discord, that's something for a later, 'proper' version of this. For now I want the minimum to keep the program running as long as possible, rather than throwing a fresh, un-caught, exception for other reasons in the logging code itself.